AppHandling/Run-AlPipeline.ps1

```diff
@@ -1746,31 +1784,54 @@ Write-Host -ForegroundColor Yellow @'
         }
     }
 
+    $appType = switch ($true) {
+        $app { "app" }
+        $testApp { "testApp" }
+        $bcptTestApp { "bcptApp" }
+        Default { "app" }
+    }
+
+    $compilationParams = $Parameters + $CopParameters
+
+    # Run pre-compile script if specified
+    if($PreCompileApp) {
+        Write-Host "Running custom pre-compilation script..."
+
+        Invoke-Command -ScriptBlock $PreCompileApp -ArgumentList $appType, ([ref] $compilationParams)
+    }
+
     try {
-        Write-Host "`nCompiling $($Parameters.appProjectFolder)"
-        $withCops = $Parameters+$CopParameters
+        Write-Host "`nCompiling $($compilationParams.appProjectFolder)"
         if ($useCompilerFolder) {
-            $appFile = Compile-AppWithBcCompilerFolder @parameters
+            $appFile = Compile-AppWithBcCompilerFolder @compilationParams
         }
         else {
-            $appFile = Invoke-Command -ScriptBlock $CompileAppInBcContainer -ArgumentList $withCops
+            $appFile = Invoke-Command -ScriptBlock $CompileAppInBcContainer -ArgumentList $compilationParams
         }
     }
     catch {
         if ($escapeFromCops) {
             Write-Host "Retrying without Cops"
+            $compilationParams = $Parameters

```

Comment:
Won't this override any changes made to compilationParams by the pre-event?
Maybe CopParameters should be added in like 1809 instead amd compilationParams kept the same as modified by the pre-event?